trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyResource.java

```diff
@@ -67,12 +67,11 @@ public void s3Head(@Context ContainerRequest request, @Suspended AsyncResponse a
     @Path("{path:.*}")
     public void s3Head(@Context ContainerRequest request, @Suspended AsyncResponse asyncResponse, @PathParam("path") String path)
     {
-        proxyClient.proxyRequest(signingController.validateAndParseAuthorization(request, SigningServiceType.S3, Optional.empty()), request, asyncResponse, getBucket(path));
+        proxyClient.proxyRequest(signingController.validateAndParseAuthorization(request, SigningServiceType.S3, Optional.empty()), parseRequest(path, request), asyncResponse);
     }
 
-    private String getBucket(String path)
+    private ParsedS3Request parseRequest(String path, ContainerRequest request)
```

Comment:
```suggestion
    private static ParsedS3Request parseRequest(String path, ContainerRequest request)
```

Reply:
This is accessing an instance field, `serverHostName`

Reply:
actually, should we just remove this function and use `ParsedS3Request.fromRequest` directly?